src/hipscat/catalog/catalog.py

```diff
@@ -1,16 +1,16 @@
 """Container class to hold catalog metadata and partition iteration"""
 
-import json
-import os
 
 from hipscat.catalog.partition_info import PartitionInfo
+from hipscat.io import file_io, paths
 
 
 class Catalog:
     """Container class for catalog metadata"""
 
-    def __init__(self, catalog_path=None):
+    def __init__(self, catalog_path: str = None) -> None:
```

Comment:
It's a bit confusing to see `None` as the return type. Shouldn't this be an object of Catalog class in most cases?

Reply:
Yeah it is confusing, but [pep specifies this is the correct way to type an init method](https://peps.python.org/pep-0484/#the-meaning-of-annotations). I believe it's to do with the way python objects are initialized, that the `__new__` method is what actually creates an instance of the class, and the `__init__` method is called after an object is created to operate on the object.

Reply:
https://realpython.com/python-class-constructor/